ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -354,15 +354,30 @@ impl Blockstore {
     ) -> Result<()> {
         let mut index0 = self.transaction_status_index_cf.get(0)?.unwrap_or_default();
         let mut index1 = self.transaction_status_index_cf.get(1)?.unwrap_or_default();
+        let slot_in_index = |slot: Slot| -> Option<u64> {
```

Comment:
I'm not sure how worried we should to be about this case given that things should be toggling very rarely, but in one of the other functions, we do explicitly have code for a slot that could be spread across both indices:
https://github.com/solana-labs/solana/blob/c6ee69e0dcb6e952fa17716a7b241f544e37e8cd/ledger/src/blockstore.rs#L2688-L2693

Reply:
Mm, good call. I forgot that we don't hold the `active_transaction_status_index` write lock while we write statuses for the whole block, just for the statuses of a single transaction.
I'll rewrite so this closure returns `Vec<u64>`, like in your PR

Reply:
Done